src/sbt-test/sbt-avro/avscparser/build.sbt

```diff
@@ -0,0 +1,27 @@
+name := "avscparser-test"
+
+libraryDependencies ++= Seq(
+  "org.apache.avro" % "avro" % "1.10.0",
+  "org.specs2" %% "specs2-core" % "4.9.4" % Test
+)
+
+(Compile / avroSchemaParser) := {
+  class AnnotateWithArtifactSchemaParser(moduleID: ModuleID)
+      extends org.apache.avro.Schema.Parser {
+    override def parse(file: java.io.File): org.apache.avro.Schema = {
+      val schema = super.parse(file)
+      if (schema.getType == org.apache.avro.Schema.Type.RECORD) {
+        schema.addProp("com.cavorite.sbt-avro.artifact", moduleID.toString())
+      }
+      schema
+    }
+  }
+
+  () => {
+    val parser = new AnnotateWithArtifactSchemaParser(projectID.value)
+    val b = org.apache.avro.Schema.createEnum(
+        "B", null, "com.cavorite.test.avscparser", java.util.Collections.singletonList("B1"))
+    parser.addTypes(java.util.Collections.singletonMap("B", b))
```

Comment:
I think that adding the types is required if you don't want to break caching.
This also ignore the validation settings
```scala
          val types = schemaParser.get().getTypes
          val validate = avroValidate.value
          val validateDefaults = avroValidateDefaults.value
          // copy of global schemaParser to avoid race condition
          // when deprecated schemaParser is removed, just create new parser instance
          new Schema.Parser()
            .addTypes(types)
            .setValidate(validate)
            .setValidateDefaults(validateDefaults)
```

Reply:
This is a demonstration in test how I see user code - by implementing task they choose explicitly validations (or rely on default which changes from false in v1.8 to true in v1.9...), and they migrate adding types from now deprecated `schemaParser` to this task instead.